packages/components/Button/src/Button.tsx

```diff
@@ -34,19 +34,56 @@ export const buttonLookup = (layer: string, state: IPressableState, userProps: B
   );
 };
 
+export const extractOuterStylePropsAndroid = memoize((style) => {
+  const marginKeys = [
+    'margin',
+    'marginTop',
+    'marginBottom',
+    'marginLeft',
+    'marginRight',
+    'marginStart',
+    'marginEnd',
+    'marginVertical',
+    'marginHorizontal',
+    'start',
+    'end',
+    'left',
+    'right',
+    'top',
+    'bottom',
+  ];
+  const outerStyleProps = {},
+    resetProps = {};
+  marginKeys.forEach((key) => {
+    if (style && style[key]) {
+      outerStyleProps[key] = style[key];
+      resetProps[key] = 0;
+    }
+  });
+  return [
+    { ...outerStyleProps, display: style ? style.display : 'flex' },
```

Comment:
Do you need to set `display` explicitly? Shouldn't it be 'flex' by default?

Reply:
Have added this to handle the case where `display: 'none'` is used to hide a component, handling nothing sent with 'flex'

Reply:
For native it's usually better to hide a component by not rendering it rather than hiding it with display: 'none', since I don't think that guarantees that the component will be hidden from screen readers.

Reply:
Added this to make sure that if used, `display: 'none'` works as intended, without this the Outer View would still be rendered taking up space while the Inner Views would not be displayed